docs/reference/scripting/using.asciidoc

```diff
@@ -1,425 +1,448 @@
-scripts per 5 minutes by default. For ingest contexts, the default script
-compilation rate is unlimited. You can change these settings dynamically by
-setting `script.context.$CONTEXT.max_compilations_rate` eg.
-`script.context.field.max_compilations_rate=100/10m`.
-
-========================================
+    variables. <<prefer-params,Use parameters>> instead of hard-coded values to decrease compile time.
 
 [discrete]
-[[modules-scripting-short-script-form]]
-=== Short script form
-A short script form can be used for brevity. In the short form, `script` is represented
-by a string instead of an object. This string contains the source of the script.
-
-Short form:
+[[hello-world-script]]
+=== Write your first script
+<<modules-scripting-painless,Painless>> is the default scripting language
+for {es}. It is secure, performant, and provides a natural syntax for anyone
+with a little coding experience.
 
-[source,js]
-----------------------
-  "script": "ctx._source.my-int++"
-----------------------
-// NOTCONSOLE
+A Painless script is structured as one or more statements and optionally
+has one or more user-defined functions at the beginning. A script must always
+have at least one statement.
 
-The same script in the normal form:
+The {painless}/painless-execute-api.html[Painless execute API] provides the ability to
+test a script with simple user-defined parameters and receive a result. Let's
+start with a simple script and review its constituent parts.
 
-[source,js]
-----------------------
-  "script": {
-    "source": "ctx._source.my-int++"
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": { <1>
+    "source": "return 'Hello, world!';" <2>
   }
-----------------------
-// NOTCONSOLE
-
-[discrete]
-[[modules-scripting-stored-scripts]]
-=== Stored scripts
-
-Scripts may be stored in and retrieved from the cluster state using the
-`_scripts` end-point.
-
-If the {es} {security-features} are enabled, you must have the following 
-privileges to create, retrieve, and delete stored scripts:
+}
+----
+<1> `script` object
+<2> `script` source
 
-* cluster: `all` or `manage`
+We're using the Painless execute API without any URL parameters, which uses
+the `painless_test` context by default. This context optionally makes
+available a `Map` of user-defined parameters, and then returns a value as an
+`Object`.
 
-For more information, see <<security-privileges>>.
+The `script` is a standard JSON object that defines scripts under most APIs
+in {es}. This object requires `source` to define the script itself. The
+example script doesn't specify a language, so it uses Painless by default.
 
+The `source` defines the script itself. The example `source` returns a
+`String` value of `Hello, world!`, converts the string to an `Object`, and
+displays the output.
 
 [discrete]
-==== Request examples
-
-The following are examples of using a stored script that lives at
```

Comment:
The examples for creating, using, viewing and deleting a stored script are useful.  Why not keep them around?

Reply:
I'll reintegrate these examples.